plugins/dmxcontrol3/exportTests/channel-numbers.js

```diff
@@ -0,0 +1,152 @@
+    if (xmlNode.$) {
+      const indexAttributes = [`dmxchannel`, `finedmxchannel`, `ultradmxchannel`, `ultrafinedmxchannel`];
+      indexAttributes.filter(attr => attr in xmlNode.$).forEach(attr => {
+        const channelIndex = xmlNode.$[attr];
+
+        if (!(channelIndex in usedChannelRanges)) {
+          usedChannelRanges[channelIndex] = [];
+        }
+
+        if (attr === `dmxchannel`) {
+          currentChannelIndex = channelIndex;
+        }
+
+        if (parseInt(channelIndex) >= channelCount) {
+          errors.push(`${attr}=${channelIndex} is out of range (maximum: ${channelCount - 1}).`);
+        }
+      });
+
+      if (`mindmx` in xmlNode.$) {
+        // xmlNode is a capability
+
+        const mindmx = parseInt(xmlNode.$.mindmx);
+        const maxdmx = parseInt(xmlNode.$.maxdmx);
+        const range = new Range([Math.min(mindmx, maxdmx), Math.max(mindmx, maxdmx)]);
+
+        addCapability(range, currentChannelIndex);
+      }
+    }
+
+    for (const tagname of Object.keys(xmlNode)) {
+      if (tagname !== `$`) {
+        for (const child of xmlNode[tagname]) {
+          findChannels(child, currentChannelIndex);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks the given capability range and adds it to the channel's ranges.
+   * @param {!Range} range A valid Range instance (start <= end).
+   * @param {!number} channelIndex The index of the channel that contains this capability.
+   */
+  function addCapability(range, channelIndex) {
+    if (channelIndex === -1) {
+      errors.push(`Capability ${range} is not inside a channel function.`);
+    }
+    else if (range.start < 0 || range.end > 255) {
+      errors.push(`Capability ${range} in channel #${channelIndex + 1} is out of the allowed 0…255 range.`);
+    }
+    else {
+      const existingRanges = usedChannelRanges[channelIndex];
+
+      if (range.overlapsWithOneOf(existingRanges)) {
+        errors.push(`Capability ${range} in channel #${channelIndex + 1} overlaps with other capabilities.`);
+      }
+      else {
+        existingRanges.push(range);
+      }
+    }
+  }
+
+  /**
+   * Checks that:
+   * - All not NoFunction channels have been used.
+   * - No NoFunction channels have been used.
+   * - Each channel has either no capability at all or its ranges span the whole 0…255 range.
+   */
+  function checkUsedChannels() {
+    mode.channels.forEach((channel, index) => {
+      const isUsed = index in usedChannelRanges;
+      const isNoFunction = channel.type === `NoFunction`;
+
+      if (isUsed) {
+        if (isNoFunction) {
+          errors.push(`Channel ${channel.name} (#${index + 1}) of type NoFunction should be omitted.`);
+        }
+        else {
+          const mergedRanges = Range.getMergedRanges(usedChannelRanges[index]);
+
+          if (!areRangesComplete(mergedRanges)) {
+            errors.push(`Channel ${channel.name} (#${index + 1}) is missing capabilities. Used ranges: ${mergedRanges.join(`, `)}`);
```

Comment:
```suggestion
            errors.push(`Channel ${index + 1} "${channel.name}" is missing capabilities. Used ranges: ${mergedRanges.join(`, `)}`);
```

(without `#` to prevent GitHub from linking it to an issue / PR)